extensions/oidc-client-registration/runtime/src/main/java/io/quarkus/oidc/client/registration/runtime/OidcClientRegistrationImpl.java

```diff
@@ -0,0 +1,244 @@
+        Buffer buffer = Buffer.buffer(clientRegJson);
+        Uni<HttpResponse<Buffer>> response = filter(request, filters, buffer).sendBuffer(buffer)
+                .onFailure(ConnectException.class)
+                .retry()
+                .atMost(oidcConfig.connectionRetryCount)
+                .onFailure().transform(t -> {
+                    LOG.warn("OIDC Server is not available:", t.getCause() != null ? t.getCause() : t);
+                    // don't wrap it to avoid information leak
+                    return new OidcClientRegistrationException("OIDC Server is not available");
+                });
+        return response.onItem();
+    }
+
+    static private HttpRequest<Buffer> filter(HttpRequest<Buffer> request, Map<Type, List<OidcRequestFilter>> filters,
+            Buffer body) {
+        if (!filters.isEmpty()) {
+            OidcRequestContextProperties props = new OidcRequestContextProperties();
+            for (OidcRequestFilter filter : OidcCommonUtils.getMatchingOidcRequestFilters(filters,
+                    OidcEndpoint.Type.CLIENT_REGISTRATION)) {
+                filter.filter(request, body, props);
+            }
+        }
+        return request;
+    }
+
+    static private RegisteredClient newRegisteredClient(HttpResponse<Buffer> resp,
+            WebClient client, String registrationUri, OidcClientRegistrationConfig oidcConfig,
+            Map<Type, List<OidcRequestFilter>> filters) {
+        if (resp.statusCode() == 200 || resp.statusCode() == 201) {
+            JsonObject json = resp.bodyAsJsonObject();
+            LOG.debugf("Client has been succesfully registered: %s", json.toString());
+
+            String registrationClientUri = (String) json.remove(OidcConstants.REGISTRATION_CLIENT_URI);
+            String registrationToken = (String) json.remove(OidcConstants.REGISTRATION_ACCESS_TOKEN);
+
+            ClientMetadata metadata = new ClientMetadata(json.toString());
+
+            return new RegisteredClientImpl(client, oidcConfig, filters, metadata,
+                    registrationClientUri, registrationToken);
+        } else {
+            String errorMessage = resp.bodyAsString();
+            LOG.errorf("Client registeration has failed:  status: %d, error message: %s", resp.statusCode(),
+                    errorMessage);
+            throw new OidcClientRegistrationException(errorMessage);
+        }
+    }
+
+    @Override
+    public Uni<RegisteredClient> readClient(String registrationUri, String registrationToken) {
+        @SuppressWarnings("resource")
+        RegisteredClient newClient = new RegisteredClientImpl(client, oidcConfig,
+                filters, createMetadata(oidcConfig.metadata), registrationUri, registrationToken);
+        return newClient.read();
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (!closed) {
+            try {
+                client.close();
+            } catch (Exception ex) {
+            }
```

Comment:
`log.debug` here for debugging purposes?

Reply:
@gastaldi Sure :-), let me check all those catch blocks